### PR TITLE
Merge latest master to stable

### PR DESCRIPTION
**Pull request by TrystanLea**
See commits below, bug fixes thanks to @emrysr and log_location branch merge https://community.openenergymonitor.org/t/switching-emonpi-emoncms-logfile-locations-to-var-log-emon/11052

### CODE REVIEW

**Comment by borpin** [comment]
Will the change from `$log_filename` to `$log_location` break existing non-emonpi setups, requiring the settings.php file to be recreated?

File: docs/SharedLinuxHostingInstall.md [hunk 1]
```diff
@@ -24,8 +24,8 @@ Then create three folders within your emoncmsdata folder called: phpfiwa, phpfin
 5) In the emoncms app directory make a copy of default_settings.php and call it settings.php.  
 Open settings.php and enter your mysql username, password and database.  
 In the feed_settings section uncomment the datadir definitions and set them to the location of each of the feed engine data folders on your system.   
-In the 'Other settings' section, change the $log_filename location to:  
-```$log_filename = dirname(__FILE__).'/' . 'emoncms.log';```
+In the 'Other settings' section, change the $log_location location to:
```
Comment (borpin) [inline docs/SharedLinuxHostingInstall.md]: Why not add this to the settings file but with a comment?

File: docs/WindowsInstall.md [hunk 2]
```diff
@@ -79,9 +79,9 @@ In the feedsettings section uncomment the datadir defenitions and set them to th
 
 On Windows '\' must be escaped with another '\' hence the '\\'
 
-In the 'Other settings' section, change the $log_filename location to:
+In the 'Other settings' section, change the $log_location location to:
 
-    $log_filename = dirname(FILE).'\\' . 'emoncms.log';
+    $log_location = dirname(FILE);
```
Comment (borpin) [inline docs/WindowsInstall.md]: Add this to the settings file with a comment.

**Comment by TrystanLea** [comment]
> Will the change from $log_filename to $log_location break existing non-emonpi setups, requiring the settings.php file to be recreated?

No it's a non breaking change. Emoncms works regardless of whether the setting is present or the emoncms log folder is created. An error is displayed on the emoncms admin page if /var/log/emoncms is missing.

**Pull request merged**